tests/contrib/data/url2purl.json

```diff
@@ -139,5 +139,33 @@
 
   "https://crates.io/api/v1/crates/rand/0.7.2/download": "pkg:cargo/rand@0.7.2",
   "https://crates.io/api/v1/crates/clap/2.33.0/download" : "pkg:cargo/clap@2.33.0",
-  "https://crates.io/api/v1/crates/structopt/0.3.11/download" : "pkg:cargo/structopt@0.3.11"
+  "https://crates.io/api/v1/crates/structopt/0.3.11/download" : "pkg:cargo/structopt@0.3.11",
+
+  "https://github.com/TG1999/fetchcode/tree/documentation/fetchcode" : "pkg:github/tg1999/fetchcode@documentation#fetchcode",
+  "https://github.com/nexB/scancode-toolkit/tree/develop/plugins/scancode-ctags-macosx_10_9_intel" : "pkg:github/nexb/scancode-toolkit@develop#plugins/scancode-ctags-macosx_10_9_intel",
+  "https://github.com/package-url/packageurl-js/tree/master/test/data" : "pkg:github/package-url/packageurl-js@master#test/data",
+  "https://github.com/package-url/packageurl-js/tree/master" : "pkg:github/package-url/packageurl-js@master",
+  "https://github.com/package-url/packageurl-js" : "pkg:github/package-url/packageurl-js",
+  "https://github.com/TG1999":null,
+  "https://github.com/TG1999/fetchcode/tree":null,
+  "https://github.com/TG1999/fetchcode/master": null,
+  "https://github.com/TG1999/fetchcode/fetchcode":null,
+
+  "https://bitbucket.org/TG1999/first_repo/src/qa/":"pkg:bitbucket/tg1999/first_repo@qa",
+  "https://bitbucket.org/TG1999/first_repo/":"pkg:bitbucket/tg1999/first_repo",
+  "https://bitbucket.org/TG1999/":null,
+  "https://bitbucket.org/TG1999/first_repo/src/":"pkg:bitbucket/tg1999/first_repo",
+  "https://bitbucket.org/TG1999/first_repo/src/master/new_folder/":"pkg:bitbucket/tg1999/first_repo@master#new_folder",
+  "https://bitbucket.org/TG1999/first_repo/new_folder/":null,
+
+  "https://gitlab.com/TG1999/firebase/-/tree/1a122122/views":"pkg:gitlab/tg1999/firebase@1a122122#views",
+  "https://gitlab.com/tg1999/firebase":"pkg:gitlab/tg1999/firebase",
+  "https://gitlab.com/TG1999/firebase/-/":null,
+  "https://gitlab.com/TG1999/firebase/-/tree":null,
+  "https://gitlab.com/TG1999/firebase/-/master":null,
+  "https://gitlab.com/TG1999/firebase/tree":null,
+  "https://gitlab.com/TG1999/firebase/master":null,
+  "https://gitlab.com/TG1999/firebase/-/tree/master":"pkg:gitlab/tg1999/firebase@master",
```

Comment:
None of these should be null

Reply:
These URLs doesn't point anywhere, they all return 404, that's why I return null, so what say :D